samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -281,83 +289,115 @@ static bool subgroup_get_valid_bis_indexes_cb(const struct bt_bap_base_subgroup
 							      ? subgroup_chan_allocation
 							      : BT_AUDIO_LOCATION_MONO_AUDIO;
 		}
+		/* keep allocation for later use */
+		bis_index->audio_allocation[data.bis[i].index - 1] = data.bis[i].chan_allocation;
+		printk("bis_index->audio_allocation[%u] = 0x%08x\n", data.bis[i].index - 1,
+		       data.bis[i].chan_allocation);
 	}
 
 	/* Get the BIS indexes */
 	for (uint8_t i = 0U; i < data.cnt; i++) {
 		if ((data.bis[i].chan_allocation == CONFIG_TARGET_BROADCAST_CHANNEL) ||
-		    ((data.bis[i].chan_allocation & CONFIG_TARGET_BROADCAST_CHANNEL) ==
-		     CONFIG_TARGET_BROADCAST_CHANNEL)) {
-			/* Exact match */
+		    (CONFIG_TARGET_BROADCAST_CHANNEL == BT_AUDIO_LOCATION_MONO_AUDIO)) {
+			/* Exact match (accept all allocations as mono) */
```

Comment:
I wonder if this is the correct behavior:

e.g. if a BASE has 2 subgroups:
Subgroup 0: LEFT and RIGHT in 1 BIS
Subgroup 1: Mono (or None)

Then it seems that when `BT_BAP_BIS_SYNC_NO_PREF` is requested, then it would simply pick the first subgroup. Wouldn't Subgroup 1 be the "more correct" choice?